### ISSUE TITLE
[padding-line-between-statements] Add an option for padding object elements

### ISSUE THREAD

**Issue by mat813** [eslint/eslint#12948]
**What rule do you want to change?**

`padding-line-between-statements`

**Does this change cause the rule to produce more or fewer warnings?**

More warnings, if enabled. (but none if --fix is enabled.)

**How will the change be implemented? (New option, new default behavior, etc.)?**

New option for STATEMENT_TYPE for object elements.

**Please provide some example code that this change will affect:**

```js
const obj = {
  props: {
    elt: Object,
  },
  data() {
    return {
      selected: {},
      config: {
        card_mode: false
      },
    };
  },
  created() {
    this.today = new Date;
  },
  methods: {
    onSubmit() {
      const url = '/foo';
      this.POST(url, {
        selected: this.selected,
      });
    },
    onReset() {
      this.$nextTick(() => this.$bvModal.hide('modal'));
    },
  },
};
```

**What does the rule currently do for this code?**

Nothing

**What will the rule do after it's changed?**

Add padding between object elements this:

```js
const obj = {
  props: {
    elt: Object,
  },

  data() {
    return {
      selected: {}

      config: {
        card_mode: false
      },
    };
  },

  created() {
    this.today = new Date;
  },

  methods: {
    onSubmit() {
      const url = '/foo';
      this.POST(url, {
        selected: this.selected,
      });
    },

    onReset() {
      this.$nextTick(() => this.$bvModal.hide('modal'));
    },
  },
};
```

**Are you willing to submit a pull request to implement this change?**

Nooooo.

**Comment by mdjermanovic** [comment]
Hi @mat813, thanks for this proposal!

`padding-line-between-statements` is about statements, so it probably shouldn't apply to elements in object literals.

Some of the alternatives might be:

* New rule for object literals only, similar to how we already have [lines-between-class-members](https://eslint.org/docs/rules/lines-between-class-members) for classes.
* New rule which would replace existing padding rules, as proposed in [this comment](https://github.com/eslint/eslint/issues/11129#issuecomment-536327392) by @kaicataldo.
* New option (or options) in [object-property-newline](https://eslint.org/docs/rules/object-property-newline).

I'm not sure which of these would be the best choice at the moment.

**Comment by mat813** [comment]
I am not sure either, but I am not sure a "one rule fits all" (2nd option) is a great idea. It would need to be very configurable, and I feel `padding-line-between-statements` already has so many options and cases to handle, an even more generic rule would be even harder to grasp and use. While I agree that it would be more concise to have only one rule, the drawback I see is that its configuration would be even larger.

When I started looking into enforcing whitespace in the code to make it breathe, and be more readable, I saw `lines-between-class-members`, and thought it was what I needed because I think I failed to grasp that there was a difference between objects and classes, but it turns out I don't have classes, and objects and classes are not the same... :-)

So, I don't know either which would be better between 1) and 3). (As 2 does not exist at the moment.)
